music21/expressions.py

```diff
@@ -517,21 +519,205 @@ def fillListOfRealizedNotes(
         fillObjects.append(firstNote)
         fillObjects.append(secondNote)
 
+    def resolveOtherPitches(self, srcPitch: pitch.Pitch):
+        # Only does something for ornament types that have "other pitches"
+        # (Turn, GeneralMordent, Trill)
+        return
+
+    def updateAccidentalDisplay(
+        self,
+        *,
+        pitchPast: list[pitch.Pitch] | None = None,
+        pitchPastMeasure: list[pitch.Pitch] | None = None,
+        otherSimultaneousPitches: list[pitch.Pitch] | None = None,
+        alteredPitches: list[pitch.Pitch] | None = None,
+        cautionaryPitchClass: bool = True,
+        cautionaryAll: bool = False,
+        overrideStatus: bool = False,
+        cautionaryNotImmediateRepeat: bool = True,
+        lastNoteWasTied: bool = False,
+    ):
+        '''
+        Updates accidental display for Ornaments with accidental(s): turns, mordents, and trills.
+        '''
+        if isinstance(self, Turn):
+            # upper first
+            p = self.upperPitch
+            if p is not None:
+                if self.upperAccid is not None:
```

Comment:
self.upperAccid is undefined for Ornament -- this is why it needs to go in Turn.

Reply:
Done